schema.sql

```diff
@@ -1,11 +1,50 @@
 CREATE EXTENSION IF NOT EXISTS "uuid-ossp";
 
+CREATE TABLE courses (
+  course_id serial PRIMARY KEY,
+  course_code varchar(5) NOT NULL,
+  course_name text
+);
+
+CREATE TABLE tracks (
+  track_id serial PRIMARY KEY,
+  track_code varchar(2) NOT NULL,
+  track_name text
+);
+
+CREATE TABLE timezones (
+  tzone_id serial PRIMARY KEY,
+  tzone_code varchar(5) NOT NULL,          -- e.g EST, CET, MIT,
+  tzone_name text,                         -- e.g Eastern Standard Time, Central European Time, Marquesas Islands Time
+  gmtoffset numeric(4,2) NOT NULL          -- e.g -5, 1, -9.50
+);
+
+
 CREATE TABLE users (
-  pkey UUID NOT NULL DEFAULT uuid_generate_v1(),
-  firstname text NOT NULL CHECK (LENGTH(TRIM(FROM firstname)) > 0),
+  user_id serial, -- change datatype from serial to uuid DEFAULT uuid_generate_v4 ()
```

Comment:
Why not start with uuid?

Reply:
Also, for consistency, we should maybe just call this `id`

Reply:
For testing purposes I used serial (serial starts at 1), so I can easily link other tables. If I started with uuid the starting value would be a long random string so when I try to reference another table I'd need to consult this value on the `users` table. This way, it always starts by 1, even if I delete the table and start adding values from scratch. Long story short: The idea is to use uuid but for testing is easier to use serial; that's why I put the commend on the code.

The reason to use `user_id` instead of `id` is because I use the same format in all other tables. I think is clearer this way.

Reply:
Ok got it. 

I wasn't sure, but just tested locally and looks like you can manually insert a uuid of your choosing, e.g.: 

```
lspeers=# INSERT INTO users (pkey, firstname, lastname) VALUES ('44f0ebfa-5e8d-11ec-9a5c-48bf6be9fc6b', 'Dmitry', 'Munchkin');
INSERT 0 1
lspeers=# select * from users;
                 pkey                 | firstname |  lastname  |         created_at
--------------------------------------+-----------+------------+----------------------------
 44f0ebfa-5e8d-11ec-9a5c-48bf6be9fc6a | Alex      | Supertramp | 2021-12-16 08:28:58.145763
 44f0ebfa-5e8d-11ec-9a5c-48bf6be9fc6b | Dmitry    | Munchkin   | 2021-12-19 11:37:10.470069
```

So wondering if we should just change to that to avoid having two columns, one for testing, one for production?

Reply:
About `id` vs. `user_id`, agree it's clear but it's might also be a bit redundant since we know we're in the `users` table. This would be similar to having a `user_firstname` column instead of `firstname`. What do you think? I'll see if I can find a source for best practice

Reply:
Lots of disagreement haha: https://stackoverflow.com/questions/208580/naming-of-id-columns-in-database-tables

It's also not as straightforward as I thought when I wrote ^. Happy to go with whatever you prefer.

Reply:
Re. `id` vs `user_id`: Yes, I can see all the disagreement around the topic, but generally speaking it seems that using just `id` is better practice. Some people even recommend naming the `id` column as `pk`. When I created the schema I was convinced that `user_id` was clearer, but to be honest I don't see know a strong reason for that... and also, the convention used by LS is `id` only, so I'd stick to that. I'll go ahead and change them.